src/pages/Admin/Users/Add/index.jsx

```diff
@@ -0,0 +1,320 @@
+ modify it under the terms of the GNU General Public License
+ version 2 as published by the Free Software Foundation.
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License along
+ with this program; if not, write to the Free Software Foundation, Inc.,
+ 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+*/
+
+import React, { useEffect, useState } from "react";
+import Title from "components/Title";
+
+// widgets
+import { Alert, Button, InputContainer, Spinner } from "components/Widgets";
+
+// services
+import { getAllFolders } from "services/folders";
+
+// predefined common fields components
+import CommonFields from "components/Upload/CommonFields";
+
+// constants
+import {
+  accessLevels,
+  bucketPool,
+  initialAddUserData,
+} from "constants/constants";
+
+const AddUser = () => {
+  // initial state to keep a track of the checkbox values
+  const [checkboxes, setCheckboxes] = useState({
+    enote: true,
+    Check_agent_copyright: false,
+    Check_agent_ecc: false,
+    Check_agent_keyword: false,
+    Check_agent_mimetype: false,
+    Check_agent_monk: false,
+    Check_agent_nomos: false,
+    Check_agent_ojo: false,
+    Check_agent_pkgagent: false,
+    Check_agent_reso: false,
+    Check_agent_shagent: false,
+  });
+
+  const initialMessage = {
+    type: "success",
+    text: "",
+  };
+
+  // initializing all the local states
+  const [addUserData, setAddUserData] = useState(initialAddUserData);
+  const [folderlist, setFolderlist] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [message, setMessage] = useState(initialMessage);
+  const [showMessage, setShowMessage] = useState(false);
+
+  // API call to add user
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    // preparing the final user data to send to the server
+    const finalUserData = { ...addUserData, public: addUserData.accessLevel };
+
+    delete finalUserData.accessLevel;
+    finalUserData.Check_agent_copyright = checkboxes.Check_agent_copyright;
+    finalUserData.Check_agent_ecc = checkboxes.Check_agent_ecc;
+    finalUserData.Check_agent_keyword = checkboxes.Check_agent_keyword;
+    finalUserData.Check_agent_mimetype = checkboxes.Check_agent_mimetype;
+    finalUserData.Check_agent_monk = checkboxes.Check_agent_monk;
+    finalUserData.Check_agent_nomos = checkboxes.Check_agent_nomos;
+    finalUserData.Check_agent_ojo = checkboxes.Check_agent_ojo;
+    finalUserData.Check_agent_pkgagent = checkboxes.Check_agent_pkgagent;
+    finalUserData.Check_agent_reso = checkboxes.Check_agent_reso;
+    finalUserData.Check_agent_shagent = checkboxes.Check_agent_shagent;
+    finalUserData.enote = checkboxes.enote ? "y" : "n";
```

Comment:
Again same thing. Can use `true/false` instead of `y/n`.

Reply:
Okay, will change it!